src/Inkluzitron/Services/PointsService.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            if (channel is not SocketGuildChannel)
+                return Task.CompletedTask; // Only server messages increments points.
+
+            return IncrementAsync(reaction);
+        }
+
+        public async Task IncrementAsync(SocketReaction reaction)
+        {
+            var user = reaction.User.IsSpecified ? reaction.User.Value : await DiscordClient.Rest.GetUserAsync(reaction.UserId);
+
+            using var context = DatabaseFactory.Create();
+            var userEntity = await GetOrCreateUserEntityAsync(context, user.Id);
+
+            if (!CanIncrementPoints(userEntity, true))
+                return;
+
+            userEntity.LastReactionPointsIncrement = DateTime.UtcNow;
+            userEntity.Points += ThreadSafeRandom.Next(0, 10);
+            await context.SaveChangesAsync();
+        }
+
+        public async Task IncrementAsync(SocketMessage message)
+        {
+            using var context = DatabaseFactory.Create();
+            var userEntity = await GetOrCreateUserEntityAsync(context, message.Author.Id);
+
+            if (!CanIncrementPoints(userEntity, false))
+                return;
+
+            userEntity.LastMessagePointsIncrement = DateTime.UtcNow;
+            userEntity.Points += ThreadSafeRandom.Next(0, 25);
+            await context.SaveChangesAsync();
+        }
+
+        static private bool CanIncrementPoints(User userEntity, bool isReaction)
+        {
+            var lastIncrement = isReaction ? userEntity.LastReactionPointsIncrement : userEntity.LastMessagePointsIncrement;
+            if (lastIncrement == null)
+                return true;
+
+            var limit = isReaction ? 0.5f : 1.0f;
+            return (DateTime.UtcNow - lastIncrement.Value).TotalMinutes >= limit;
+        }
+
+        static private async Task<User> GetOrCreateUserEntityAsync(BotDatabaseContext context, ulong userId)
+        {
+            var userEntity = await context.Users.AsQueryable().FirstOrDefaultAsync(o => o.Id == userId);
+
+            if (userEntity != null)
+                return userEntity;
+
+            userEntity = new User() { Id = userId };
+            await context.AddAsync(userEntity);
+
+            return userEntity;
+        }
+
+        static private async Task<int> CalculatePositionAsync(BotDatabaseContext context, IUser user)
+        {
+            var users = await context.Users.AsQueryable()
+                .Where(o => o.Points > 0)
+                .ToListAsync();
+
+            return users.FindIndex(o => o.Id == user.Id) + 1;
+        }
+
+        public async Task<SysDraw.Image> GetPointsAsync(IUser user)
+        {
+            using var context = DatabaseFactory.Create();
+            var userEntity = await context.Users.AsQueryable().FirstOrDefaultAsync(o => o.Id == user.Id);
+
+            if (userEntity == null)
+                return null;
+
+            const string font = "DejaVu Sans";
+            using var positionFont = new Font(font, 45F);
```

Comment:
Podle všecho se tohle nemění, co z toho udělat properties, ať se to zbytečně nevytváří při každém vykreslování?